syntax = "proto3";
import "status.proto";
package adsmsg;

message NonBibRecord {
    string bibcode = 1;
    bool refereed = 2;
    repeated string simbad_objects = 3;
    repeated string grants = 4;
    repeated string citations = 5;
    float boost = 6;
    int32 citation_count = 7;
    int32 read_count = 8;
    repeated string readers = 9;
    repeated int32 downloads = 10;
    repeated int32 reads = 11;
    repeated string reference = 12;
    repeated string ned_objects = 13;
    // data field is array of sub types:count: ['NED:15', 'CDS:5']
    repeated string data = 14;
    // data_counts holds the sum of all the counts in the data map
    int32 data_count = 15;
    map<string, DataLinksValue> data_links = 16;
    // data link properties
    repeated string esource = 17;
    repeated string property = 18;
    Status status = 19;
}    

message DataLinksValue {
	repeated string value = 1;
}

message NonBibRecordList {
	repeated NonBibRecord nonbib_records = 1;
	Status status = 2;
}
